tidb-lightning/tidb-lightning-configuration.md

```diff
@@ -8,462 +8,675 @@ aliases: ['/docs/dev/tidb-lightning/tidb-lightning-configuration/','/docs/dev/re
+- Range: `[0, 1)`
+
+<!-- Example: `0.75` -->
+
+#### `data-source-dir`
+
+- Specifies the local source data directory or the URI of the external storage. For more information about the URI of the external storage, see [URI format](/br/backup-and-restore-storages.md#uri-format).
+
+<!-- Example: `"/data/my_database"` -->
+
+#### `character-set`
+
+- Specifies the character set of the schema files, containing `CREATE TABLE` statements.
+- Default value: `"auto"`
+- Value options:
+    - `"auto"`: automatically detects whether the schema is UTF-8 or GB-18030. An error is reported if the encoding is neither.
+    - `"utf8mb4"`: the schema files must be encoded as UTF-8; otherwise, an error is reported.
+    - `"gb18030"`: the schema files must be encoded as GB-18030; otherwise, an error is reported
+    - `"latin1"`:  the schema files use MySQL latin1 encoding, also known as Code Page 1252.
+    - `"binary"`:  do not try to decode the schema files
+
+#### `data-character-set`
+
+- Specifies the character set of the source data file. TiDB Lightning converts the source file from the specified character set to UTF-8 encoding when importing.
+- Currently, this configuration only specifies the character set of the CSV files with the following options supported. If left blank, the default value `"binary"` is used, that is to say, Lightning does not convert the encoding.
+- TiDB Lightning does not predict about the character set of the source data file and only converts the source file and import the data based on this configuration.
+- If the value of this configuration is not the same as the actual encoding of the source data file, a failed import, data loss or data disorder might appear.
+- Default value: `"binary"`
+- Value options:
+    - `"binary"`: Indicates that Lightning does not convert the encoding (by default).
+    - `"utf8mb4"`: Indicates that the source data file uses UTF-8 encoding.
+    - `"GB18030"`: Indicates that the source data file uses the GB-18030 encoding.
+    - `"GBK"`: The source data file uses GBK encoding (GBK encoding is an extension of the GB-2312 character set, also known as Code Page 936).
+    - `"latin1"`: The source data file uses MySQL latin1 encoding, also known as Code Page 1252.
+
+#### `data-invalid-char-replace`
+
+- Specifies the replacement character in case of incompatible characters during the character set conversion of the source data file.
+- This configuration must not be duplicated with field separators, quote definers, and line breaks. Changing the default value might result in potential degradation of parsing performance for the source data file.
+- Default value: `"\uFFFD"`, which is the "error" Rune or Unicode replacement character in UTF-8 encoding.
+
+#### `strict-format`
+
+- Specifies the input data in a [strict format](/tidb-lightning/tidb-lightning-data-source.md#strict-format) speeds up processing. The default value is false for safety instead of speed.
```

Comment:
```suggestion
- Specifies the input data in a [strict format](/tidb-lightning/tidb-lightning-data-source.md#strict-format) to speed up processing. The default value is `false` for safety instead of speed.
```